src/structures/Channel.js

```diff
@@ -108,19 +108,59 @@ class Channel extends Base {
   }
 
   /**
-   * Indicates whether this channel is {@link TextBasedChannels text-based}.
+   * Indicates whether this channel is a {@link TextChannel}.
    * @returns {boolean}
    */
   isText() {
-    return 'messages' in this;
+    return this.type === 'GUILD_TEXT';
```

Comment:
```suggestion
    return this.type === ChannelTypes.GUILD_TEXT;
```

Reply:
I'm all for enum usages but this would never return true... pretty sure that ChannelTypes.GUILD_TEXT is a number, not a string

Reply:
I tested this and you're right. I will change it to ```ChannelTypes[ChannelTypes.X]``` for now but perhaps a function for this would be a little more subtle

Reply:
I mean, to be fair using an indexed access here seems like a lot more verbose, and it (maybe...?) has performance implications rather than using a string comparison.

Reply:
All (afaik) true, but i still believe that raw strings are not the way to go. If discord decides to rename channel types for some reason (especially since they work with int numbers so all good), it gets messy. We should try and minimize the exposure of raw strings all around the project and gather them in the least possible files needed. This way, something changes, you change it once.

Reply:
If the type name changed you’d be having to change the name of the method most likely anyway. Isn’t this solution much worse performance wise? And I mean you still have the channel type string there so it doesn’t improve anything in that regard